packages/autorest.go/src/transform/namer.ts

```diff
@@ -100,23 +100,69 @@ export async function namer(session: Session<CodeModel>) {
   model.language.go!.singleClient = singleClient;
 
   // fix up type names
-  const structNames = new Set<string>();
+  // final names are added to typeNames which is used to detect collisions
+  const typeNames = new Set<string>();
```

Comment:
Is this just `names` now that there are consts in it too?

Reply:
Pretty much, but I felt names was a little too generic. It's really `structAndConstNames` but I went with the shorter `typeNames`.